search/filters/ExactMatchFilter.php

```diff
@@ -50,6 +50,7 @@ protected function applyOne(DataQuery $query) {
 	 */
 	protected function applyMany(DataQuery $query) {
 		$this->model = $query->applyRelation($this->relation);
+		$column = $query->lastAlias ? sprintf('"%s"."%s"', $query->lastAlias, $this->name) : $this->getDbName();
```

Comment:
Ok, how about this….

Since one class can have multiple aliases, but not the other way around, applyRelation returns the relation name alias and you can GET the class by $dataQuery->relationClass($alias); 

This would have to be done in master though, and in 3.x we could stick with the property (lastAlias, although maybe stick it in a getter?) and mark it as deprecated in 4.